.test-infra/jenkins/dependency_check/dependency_check_report_generator.py

```diff
@@ -105,18 +108,26 @@ def prioritize_dependencies(deps, sdk_type):
     try:
       logging.info("\n\nStart processing: " + dep)
       dep_name, curr_ver, latest_ver = extract_single_dep(dep)
-      curr_release_date, latest_release_date = query_dependency_release_dates(bigquery_client,
-                                                                              dep_name,
-                                                                              curr_ver,
-                                                                              latest_ver)
+      curr_release_date = None
+      latest_release_date = None
       group_id = None
+
       if sdk_type == 'Java':
         # extract the groupid and artifactid
         group_id, artifact_id = dep_name.split(":")
         dep_details_url = "{0}g:\"{1}\" AND a:\"{2}\"".format(_MAVEN_CENTRAL_URL, group_id, artifact_id)
+        curr_release_date = find_release_time_from_maven_central(group_id, artifact_id, curr_ver)
+        latest_release_date = find_release_time_from_maven_central(group_id, artifact_id, latest_ver)
       else:
         dep_details_url = _PYPI_URL + dep_name
-
+        curr_release_date = find_release_time_from_python_compatibility_checking_service(dep_name, curr_ver)
+        latest_release_date = find_release_time_from_python_compatibility_checking_service(dep_name, curr_ver)
+
+      if not curr_release_date or not latest_release_date:
+        curr_release_date, latest_release_date = query_dependency_release_dates_from_bigquery(bigquery_client,
```

Comment:
So we are going to keep BQ around as a backup ?

Reply:
Yes. 
1. Need the bigquery table to record release dates of non-MavenCentral packages.
2. The Python compatibility checking services is not mature, 10% of requests will timeout even run out 3 retries.